inloop/solutions/views.py

```diff
@@ -52,8 +53,11 @@ def post(self, request, slug):
             messages.error(request, "You haven't uploaded any files.")
             return failure_response
 
-        if not all([file_name.endswith(".java") for file_name, _ in uploads.items()]):
-            messages.error(request, "You have uploaded invalid files (allowed: *.java).")
+        if not all([validate(file_name) for file_name, _ in uploads.items()]):
+            endings = ", ".join(parse_allowed_file_endings())
+            messages.error(
+                request, "You have uploaded invalid files (allowed: {}).".format(endings)
+            )
```

Comment:
How about creating a `validate_upload` method that raises an exception with the message? This helps hiding the implementation details in the validators module.

```python
from django.core.validators import ValidationError
# [...]
try:
    validate_filenames(uploads)
except ValidationError as e:
    messages.error(request, e.message)
    return redirect_to_upload
```

Reply:
I agree, but since we now have two possibilities to upload files (editor and upload panel), we should use `failure_response` instead of `redirect_to_upload` in this particular case.